src/transformers/models/clip/configuration_clip.py

```diff
@@ -0,0 +1,266 @@
+            If True, use gradient checkpointing to save memory at the expense of slower backward pass.
+
+        Example::
+
+        >>> from transformers import CLIPVisionModel, CLIPTextConfig
+
+        >>> # Initializing a CLIPVisionModel with valhalla/clip-vit-base-patch32 style configuration
+        >>> configuration = CLIPConfig()
+
+        >>> # Initializing a model from the valhalla/clip-vit-base-patch32 style configuration
+        >>> model = CLIPVisionModel(configuration)
+
+        >>> # Accessing the model configuration
+        >>> configuration = model.config
+    """
+
+    model_type = "clip_vision_model"
+
+    def __init__(
+        self,
+        hidden_size=768,
+        intermediate_size=3072,
+        num_hidden_layers=12,
+        num_attention_heads=12,
+        image_size=224,
+        patch_size=32,
+        layer_norm_eps=1e-5,
+        dropout=0.0,
+        attention_dropout=0.0,
+        initializer_range=0.02,
+        initializer_factor=1.0,
+        gradient_checkpointing=False,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        self.hidden_size = hidden_size
+        self.intermediate_size = intermediate_size
+        self.dropout = dropout
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.patch_size = patch_size
+        self.image_size = image_size
+        self.initializer_range = initializer_range
+        self.initializer_factor = initializer_factor
+        self.attention_dropout = attention_dropout
+        self.layer_norm_eps = layer_norm_eps
+        self.gradient_checkpointing = gradient_checkpointing
+
+
+class CLIPConfig(PretrainedConfig):
+    r"""
+    :class:`~transformers.CLIPConfig` is the configuration class to store the configuration of a
+    :class:`~transformers.CLIPModel`. It is used to instantiate CLIP model according to the specified arguments,
+    defining the text model and vision model configs.
+
+    Configuration objects inherit from :class:`~transformers.PretrainedConfig` and can be used to control the model
+    outputs. Read the documentation from :class:`~transformers.PretrainedConfig` for more information.
+
+    Args:
+        projection_dim: (:obj:`int`, `optional`, defaults to 512):
+            Dimentionality of text and vision projection layers.
+        kwargs (`optional`):
+            Dictionary of keyword arguments. Notably:
+
+                - **text_config** (:class:`~transformers.CLIPTextConfig`, `optional`) -- An instance of a configuration
+                  object that defines the text model config.
+                - **vision_config** (:class:`~transformers.CLIPVisionConfig`, `optional`) -- An instance of a
+                  configuration object that defines the vision model config.
+    """
+
+    model_type = "clip"
+    is_composition = True
+
+    def __init__(self, projection_dim=512, **kwargs):
```

Comment:
```suggestion
    def __init__(self, text_config_dict, vision_config_dict, projection_dim=512, **kwargs):
```

IMO, it should not be possible to instantiate a `CLIPConfig` without passing the two configs

Reply:
or add an assert that both configs have to be in kwargs